### PR TITLE
fix(#1961/#2958): Manage focus in docs popover breadcrumbs

### LINKED ISSUE #1961: Manage focus in docs popover breadcrumbs
# 🐛 Bug Report 

When a link within a popover changes the contents of the popover, focus will be lost to the document.body, because the element that had focus will be removed from the DOM and replaced with new content.
There was prior work done on this, but we had to change some CSS to do it, we should figure out a way that doesn't require any CSS changes to the component. So this is really about managing the focus in our docs, not necessarily about Breadcrumbs themselves.
https://github.com/adobe/react-spectrum/pull/728

## 🤔 Expected Behavior

Focus should not go to the body.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |
| Browser          |
| Operating System |

### PR DESCRIPTION
**Pull request by majornista**
When a link within a popover changes the contents of the popover, focus will be lost to the document.body, because the element that had focus will be removed from the DOM and replaced with new content.

There was prior work done on this, but we had to change some CSS to do it, we should figure out a way that doesn't require any CSS changes to the component. So this is really about managing the focus in our docs, not necessarily about Breadcrumbs themselves.
#728

We can make changes to FocusScope to better account for a content changes within the Focus resulting in a loss of focus while the scope remains active, as when the contents of a Dialog change due to actions taken within the Dialog, but for the Breadcrumbs use case, for a consistent user experience, it seems that the current BreadcrumbItem still needs to be focused explicitly.

Closes #1961 

Related issue/PR:  #2592/#2594 This related PR addresses the problem when the current breadcrumb changes from an action within the Breadcrumbs, but it can't account for explicitly focusing the current BreadcrumbItem after a content change initiated from elsewhere. I think we still have to handle that within our docs.

## ✅ Pull Request Checklist:

- [x] Included link to corresponding React Spectrum GitHub Issues #1961 and #2958.
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

1. Open https://reactspectrum.blob.core.windows.net/reactspectrum/99eb47497ee388f61e737fded0fcdfb8a0433bbe/docs/react-aria/useMenu.html
2. Tab to `AriaMenuOptions` link in the [API code example](https://reactspectrum.blob.core.windows.net/reactspectrum/99eb47497ee388f61e737fded0fcdfb8a0433bbe/docs/react-aria/useMenu.html#api).
3. Press Enter to expand `AriaMenuOptions` popover.
4. BreadcrumbItem for the current location, `AriaMenuOptions`, should have focus 
5. Drill down to AriaMenuOptions > CollectionChildren > CollectionElement > SectionElement > SectionProps > ItemElement > ItemProps. The dropdown menu should display. After clicking each link to refresh the contents of the popover, the  BreadcrumbItem for the current location should automatically focus.
6. Open dropdown and arrow down to focus, then press space to activate CollectionElement.
CollectionElement BreadcrumbItem element should autoFocus if things are working correctly, otherwise focus will be lost to the document.body, and the next tab stop will be and element outside of the popover.

## 🧢 Your Project:

Adobe/Accessibility

### CODE REVIEW

**Review by ktabors — COMMENTED** [review]
Can you provide more testing instructions? When I compare this to the existing version I see the additional aria-haspopup and aria-label, but I'm not able to see a change in behavior.

**Comment by majornista** [comment]
> Can you provide more testing instructions? When I compare this to the existing version I see the additional aria-haspopup and aria-label, but I'm not able to see a change in behavior.

@ktabors Sorry about that. I just realized that this PR is somewhat dependent on #2594, which adds `tabIndex = -1`, to the current breadcrumb item at https://github.com/adobe/react-spectrum/pull/2594/files#diff-e657bc4a0257fee85f0ddd4f6ba8722d3ffb177175232bbb97c3dde539ee6a6bR45 so that it will always be focusable but not tabbable. Should be fixed by https://github.com/adobe/react-spectrum/pull/2749/commits/e2b911146395a7fc1f3006bda48dbde4881d72fc.

**Review by LFDanLu — DISMISSED** [review]
LGTM, verified the new behavior. This change should be fine IMO since it is limited to the docs

**Review by devongovett — APPROVED** [review]
It's ok as a temporary solution for the docs, but I think we should really come up with a nicer solution. I don't think we should recommend manually mutating the DOM to users of React Spectrum.

**Comment by majornista** [comment]
@LFDanLu and @devongovett Note that I’ve added https://github.com/adobe/react-spectrum/issues/2958 to document the addition of `autoFocusCurrent`  prop for Breadcrumbs.

File: packages/@react-aria/focus/src/useFocusable.tsx [hunk 1]
```diff
@@ -73,10 +74,10 @@ export function useFocusable(props: FocusableOptions, domRef: RefObject<HTMLElem
 
   useEffect(() => {
     if (autoFocusRef.current && domRef.current) {
-      domRef.current.focus();
+      focusSafely(domRef.current);
```
Comment (majornista) [inline packages/@react-aria/focus/src/useFocusable.tsx:77]: Use focusSafely, so auto-focusing an element in a popover does not cause the popover to immediately close.

**Review by devongovett — APPROVED** [review]

**Pull request merged**